First draft of stemcell verification pipeline

zhanggbj opened the pull request:
[#130342357](https://www.pivotaltracker.com/story/show/130342357)

Commit:
First draft of stemcell verification pipeline

[#130342357](https://www.pivotaltracker.com/story/show/130342357)

maximilien commented on ci/tasks/cf-push-app.yml:
```diff
@@ -0,0 +1,17 @@
+  path: gopath/src/github.com/cloudfoundry-community/bosh-softlayer-tools/ci/tasks/cf-push-app.sh
+
+params:
+  CF-API:                 replace-me
```
Why these extra spaces in YML... I'd remove them since as you know YML is sensitive to spaces and having them creates potential issue later. So section below should be like:

``` yml
CF-USERNAME: replace-me
 CF-PASSWORD: replace-me
 APP-API: replace-me
 CF-NAMESERVER: replace-me 
```

zhanggbj replied:
sure, will update this in next commit.

maximilien commented on ci/tasks/upgrade-stemcell.sh:
```diff
@@ -0,0 +1,92 @@
+
+print_title "INSTALL BOSH CLI..."
+echo "installing bosh CLI"
+gem install bosh_cli --no-ri --no-rdo c
```
Can we start preparing for the new BOSH CLI... maybe have the section that download CLI and issues CLI command be a separate script since new CLI will have different params and options. 

(not important but interesting to do if you have time)

zhanggbj replied:
I would like to have a try with new BOSH CLI, but not sure if we can switch to it at this stage, maybe after I got familiar with params and options, we can start using new CLI in our pipeline.

maximilien replied:
@zhanggbj get it here: https://github.com/cloudfoundry/bosh-cli

It's good to have it on your machine and start trying it. Best,

Max

zhanggbj replied:
Started to try v2 on my own, but didn't update it here as the director I'm using for upgrade have not upgrade to use ssl yet.  And moved bosh cli installation to utils.sh, so we can change it later. Thanks.

maximilien reviewed (changes requested):
Overall looks great. Couple minor comments and also generally I would like to see the `.sh` files smaller (modularized) if possible to make it easier to debug.

Howerver, just noting since we have a lot to do. If you agree and cannot address now, just create a chore to do it for v4 since that's also when we'd use new BOSH CLI. Thanks,

Max

Commit:
Refine upgrade verification pipeline

 1. Modularized scripts
 2. Added tasks to handle upgrade success/failure
 3. Addressed PR comments
[#130342357](https://www.pivotaltracker.com/story/show/130342357)
